### PR TITLE
fix(desks): fixed desks service to handle 200+ desks/stages.

### PR DESCRIPTION
**Pull request by eozan**
SD-1882

### CODE REVIEW

**Comment by amagdas** [comment]
:+1:

**Comment by mugurrus** [comment]
:+1:

File: client/app/scripts/superdesk-desks/desks.js [hunk 1]
```diff
@@ -275,6 +275,21 @@
 
                 var userDesks, userDesksPromise;
 
+                var _fetchCombined = function(endpoint, page, items) {
```
Comment (mugurrus) [inline client/app/scripts/superdesk-desks/desks.js]: This retrieves the whole collection. Shouldn't there be a limit though?

**Comment by sivakuna-aap** [comment]
:+1: 
Probably a better name for `_fetchCombined()`
It would be better if we can externalize the `max_results`. In future the value can be decreased or increased.

File: client/app/scripts/superdesk-desks/desks.js (hunk 1 above)
Comment (eozan) [inline client/app/scripts/superdesk-desks/desks.js]: Currently, desks service fetches all desks/stages/members and compiles lookup tables to use app-wide, and supposedly this should happen only once.

Ideally it needs to change, and fetch data only when needed and cache for future use. But until then, this needed to be done to be able to display stages.

**Comment by mdhaman** [comment]
:+1:

**Comment by ioanpocol** [comment]
:+1:

**Pull request merged**